ds3-autogen-go/src/main/kotlin/com/spectralogic/ds3autogen/go/generators/response/BaseResponseGenerator.kt

```diff
@@ -0,0 +1,171 @@
+     */
+    fun toNullPayloadResponseCode(code: Int, responseName: String): ResponseCode {
+        return ResponseCode(code, "return &$responseName{}, nil")
+    }
+
+    /**
+     * Creates the Go code for parsing a response with a payload
+     */
+    fun toPayloadResponseCode(code: Int, responseName: String): ResponseCode {
+        val parseResponse = "var body $responseName\n" +
+                indent(2) + "if err := readResponseBody(webResponse, &body); err != nil {\n" +
+                indent(3) + "return nil, err\n" +
+                indent(2) + "}\n" +
+                indent(2) + "return &body, nil"
+
+        return ResponseCode(code, parseResponse)
+    }
+
+    /**
+     * Retrieves the comma separated list of expected status codes
+     */
+    fun toExpectedStatusCodes(responseCodes: ImmutableList<Ds3ResponseCode>?): String {
+        if (isEmpty(responseCodes)) {
+            return ""
+        }
+        return responseCodes!!.stream()
+                .filter { code -> code.code < ERROR_THRESHOLD }
+                .map { code -> code.code.toString() }
+                .collect(Collectors.joining(", "))
+    }
+
+    /**
+     * Retrieves the content of the response struct, which assumes a spectra defined
+     * Ds3Type.
+     */
+    override fun toResponsePayloadStruct(expectedResponseCodes: ImmutableList<Ds3ResponseCode>?, typeMap: ImmutableMap<String, Ds3Type>): String {
+        if (isEmpty(expectedResponseCodes)) {
+            throw IllegalArgumentException("Expected at least one response payload")
+        }
+        val responsePayloadType = getResponsePayload(expectedResponseCodes)
+        val responseModel = typeMap[responsePayloadType]
+
+        val responseName = removePath(responsePayloadType)
+        val responseType = toResponsePayloadType(responsePayloadType, expectedResponseCodes)
+        val xmlParsing = toXmlParsingPayload(responseModel)
+        return "$responseName $responseType `xml:\"$xmlParsing\"`"
+    }
+
+    /**
+     * Determines if a response payload is a struct or a pointer to a struct. When there
+     * are multiple response codes, then the type should be a pointer to a struct.
+     */
+    fun toResponsePayloadType(responsePayload: String, responseCodes: ImmutableList<Ds3ResponseCode>?): String {
+        if (isEmpty(responseCodes)) {
+            throw IllegalArgumentException("Expected at least one response code")
+        }
+        if (responseCodes!!.filter { code -> code.code < ERROR_THRESHOLD }.size == 1) {
+            return removePath(responsePayload)
+        }
+        return "*" + removePath(responsePayload)
+    }
+
+    /**
+     * Retrieves the xml tag to be parsed for the specified Ds3Type
+     */
+    fun toXmlParsingPayload(ds3Type: Ds3Type?): String {
+        if (ds3Type == null) {
+            throw IllegalArgumentException("No Ds3Type defined for command")
+        }
+        if (ds3Type.nameToMarshal != null && !ds3Type.nameToMarshal.equals("Data")) {
```

Comment:
I believe Kotlin allows you to do `==` with strings instead of having to call `.equals`